src/core/load-fragment.c

```diff
@@ -1591,21 +1591,30 @@ int config_parse_service_sockets(
         assert(rvalue);
         assert(data);
 
-        FOREACH_WORD_QUOTED(word, l, rvalue, state) {
-                _cleanup_free_ char *t = NULL, *k = NULL;
+        prev = cur = rvalue;
+        for(;;) {
+                _cleanup_free_ char *word = NULL, *t = NULL, *k = NULL;
 
-                t = strndup(word, l);
-                if (!t)
+                r = extract_first_word(&cur, &word, NULL, 0);
+                if (r == 0)
+                        break;
+                if (r == -ENOMEM)
                         return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Trailing garbage in sockets, ignoring: %s", prev);
```

Comment:
any specific reason it's been used in 
config_parse_bounding_set and config_parse_namespace_path_strv ?